contracts/src/SimplePDPService.sol

```diff
@@ -142,37 +134,51 @@ contract SimplePDPService is PDPListener, PDPRecordKeeper, Initializable, UUPSUp
         }
-
         if (challengeCount < getChallengesPerProof()) {
             revert("Invalid challenge count < 5");
         }
-        emit Debug("deadline", provingDeadlines[proofSetId]);
-        emit Debug("block number", block.number);
-        // check for late proof 
+        // check for proof outside of proving period
         if (provingDeadlines[proofSetId] < block.number) {
-            uint256 periodsLate = 1 + ((block.number - provingDeadlines[proofSetId]) / getMaxProvingPeriod());
-            emit Debug("we're late", periodsLate); 
-            emit FaultRecord(FaultType.LATE, periodsLate);
+            revert("Current proving period passed. Open a new proving period.");
+        } 
+        if (provingDeadlines[proofSetId] - getMaxProvingPeriod() >= block.number) {
+            revert("Too early. Wait for proving period to open");
         }
         provenThisPeriod[proofSetId] = true;
     }
 
-    // nextProvingPeriod checks for unsubmitted proof and emits a fault record if so
+    // nextProvingPeriod checks for unsubmitted proof and emits a fault if so
     function nextProvingPeriod(uint256 proofSetId, uint256 leafCount) external onlyPDPVerifier {
         receiveProofSetEvent(proofSetId, OperationType.NEXT_PROVING_PERIOD, abi.encode(leafCount));
-        // check for unsubmitted proof 
-        if (!provenThisPeriod[proofSetId]) {
-            uint256 periodsSkipped = 1;
-            if (provingDeadlines[proofSetId] < block.number) {
-                periodsSkipped = 1 + ((block.number - provingDeadlines[proofSetId]) / getMaxProvingPeriod());
+        // Noop when proving period not yet open
+        // Can only get here if calling nextProvingPeriod multiple times within the same proving period
+        uint256 prevDeadline = provingDeadlines[proofSetId] - getMaxProvingPeriod();
+        if (block.number <= prevDeadline) {
+            return;
+        }
+        // Within the open proving period
+        if (block.number <= provingDeadlines[proofSetId]) {
+            if (!provenThisPeriod[proofSetId]) {
+                emit FaultRecord(1);
             }
-            emit FaultRecord(FaultType.SKIPPED, periodsSkipped);
+            provingDeadlines[proofSetId] = provingDeadlines[proofSetId] + getMaxProvingPeriod();
+            provenThisPeriod[proofSetId] = false;
+            return;
+        }
+        // Past the open proving period
+        uint256 periodsSkipped = (block.number - provingDeadlines[proofSetId]) / getMaxProvingPeriod();
+        uint256 faultPeriods = periodsSkipped;
+        if (!provenThisPeriod[proofSetId]) { 
+            // include previous unproven period 
+            faultPeriods += 1;
+        }
+        if (faultPeriods > 0) {
+            emit FaultRecord(faultPeriods);
         }
+        provingDeadlines[proofSetId] = provingDeadlines[proofSetId] + getMaxProvingPeriod()*periodsSkipped; 
```

Comment:
Nit: I think this code could also handle the on-time case, with something like `periodsSkipped = max(0, (block.number - provingDeadlines[proofSetId])) / getMaxProvingPeriod()`. The rest is basically duplicate.